packages/pocket-ic/src/nonblocking.rs

```diff
@@ -1413,3 +1414,68 @@ fn setup_tracing(pid: u32) -> Option<WorkerGuard> {
         _ => None,
     }
 }
+
+/// Retrieves a default effective canister id for canister creation on a PocketIC instance
+/// characterized by:
+///  - a PocketIC instance URL of the form http://<ip>:<port>/instances/<instance_id>;
+///  - a PocketIC HTTP gateway URL of the form http://<ip>:port for a PocketIC instance.
+///
+/// Returns an error if the PocketIC instance topology could not be fetched or parsed, e.g.,
+/// because the given URL points to a replica (i.e., does not meet any of the above two properties).
+///
+/// Panics if the PocketIC instance topology contains no application, verified application, and system subnet.
+pub async fn get_default_effective_canister_id(
+    pocket_ic_url: String,
+) -> Result<Principal, DefaultEffectiveCanisterIdError> {
+    let client = reqwest::Client::new();
+    let res = loop {
+        let res = client
+            .get(format!(
+                "{}{}",
+                pocket_ic_url.trim_end_matches('/'),
+                "/_/topology"
+            ))
+            .send()
+            .await?;
+        if res.status() == StatusCode::CONFLICT {
+            std::thread::sleep(Duration::from_millis(POLLING_PERIOD_MS));
+        } else {
+            break res.error_for_status()?;
+        }
+    };
+    let topology_bytes = res.bytes().await?;
+    let topology_str = String::from_utf8(topology_bytes.to_vec())?;
+    let topology: Topology = serde_json::from_str(&topology_str)?;
+
+    let subnet_ids: Vec<_> = if !topology.get_app_subnets().is_empty() {
+        topology.get_app_subnets()
+    } else if !topology.get_verified_app_subnets().is_empty() {
+        topology.get_verified_app_subnets()
+    } else if !topology.get_system_subnets().is_empty() {
+        topology.get_system_subnets()
```

Comment:
Consider reordering the if-else branches so that verified application and system subnets take precedence over application subnets: newly created subnets with mainnet canister ids will be application subnets and thus if somebody starts with a PocketIC instance containing a system subnet, then the function `get_default_effective_canister_id` would switch to application subnets after an (application) subnet with mainnet canister ids is created.